source/type/tests/headsign_test.cpp

```diff
@@ -0,0 +1,153 @@
+
+    // check that stop_time headsigns are correct
+    BOOST_CHECK_EQUAL(headsign_handler.get_headsign(vj_vec[0]->stop_time_list.at(0)), "A00");
+    BOOST_CHECK_EQUAL(headsign_handler.get_headsign(vj_vec[0]->stop_time_list.at(1)), vj_vec[1]->name);
+    BOOST_CHECK_EQUAL(headsign_handler.get_headsign(vj_vec[1]->stop_time_list.at(0)), vj_vec[1]->name);
+    BOOST_CHECK_EQUAL(headsign_handler.get_headsign(vj_vec[1]->stop_time_list.at(1)), "B11");
+    BOOST_CHECK_EQUAL(headsign_handler.get_headsign(vj_vec[1]->stop_time_list.at(2)), "B11");
+    BOOST_CHECK_EQUAL(headsign_handler.get_headsign(vj_vec[2]->stop_time_list.at(0)), vj_vec[2]->name);
+    BOOST_CHECK_EQUAL(headsign_handler.get_headsign(vj_vec[2]->stop_time_list.at(1)), "C21");
+    BOOST_CHECK_EQUAL(headsign_handler.get_headsign(vj_vec[2]->stop_time_list.at(2)), vj_vec[2]->name);
+    BOOST_CHECK_EQUAL(headsign_handler.get_headsign(vj_vec[3]->stop_time_list.at(0)), vj_vec[3]->name);
+
+    // check that we can retrieve every vj from matching headsign
+    BOOST_CHECK_EQUAL(headsign_handler.get_vj_from_headsign("vehicle_journey 0").size(), 1);
+    BOOST_CHECK(navitia::contains(headsign_handler.get_vj_from_headsign(vj_vec[0]->name), vj_vec[0]));
+    BOOST_CHECK_EQUAL(headsign_handler.get_vj_from_headsign("A00").size(), 1);
+    BOOST_CHECK(navitia::contains(headsign_handler.get_vj_from_headsign("A00"), vj_vec[0]));
+    BOOST_CHECK_EQUAL(headsign_handler.get_vj_from_headsign("vehicle_journey 1").size(), 2);
+    BOOST_CHECK(navitia::contains(headsign_handler.get_vj_from_headsign(vj_vec[1]->name), vj_vec[0]));
+    BOOST_CHECK(navitia::contains(headsign_handler.get_vj_from_headsign(vj_vec[1]->name), vj_vec[1]));
+    BOOST_CHECK_EQUAL(headsign_handler.get_vj_from_headsign("B11").size(), 1);
+    BOOST_CHECK(navitia::contains(headsign_handler.get_vj_from_headsign("B11"), vj_vec[1]));
+    BOOST_CHECK_EQUAL(headsign_handler.get_vj_from_headsign(vj_vec[2]->name).size(), 1);
+    BOOST_CHECK(navitia::contains(headsign_handler.get_vj_from_headsign(vj_vec[2]->name), vj_vec[2]));
+    BOOST_CHECK_EQUAL(headsign_handler.get_vj_from_headsign("C21").size(), 1);
+    BOOST_CHECK(navitia::contains(headsign_handler.get_vj_from_headsign("C21"), vj_vec[2]));
+    BOOST_CHECK_EQUAL(headsign_handler.get_vj_from_headsign(vj_vec[3]->name).size(), 1);
+    BOOST_CHECK(navitia::contains(headsign_handler.get_vj_from_headsign(vj_vec[3]->name), vj_vec[3]));
+
+    boost::archive::text_oarchive oa(std::cout);
+    oa & b.data->pt_data;
+}
+
+struct HeadsignHandlerTest: nt::HeadsignHandler {
+    std::unordered_map<const nt::VehicleJourney*, boost::container::flat_map<uint16_t, std::string>>
+    get_map_vj_map_stop_time_headsign_change() {return map_vj_map_stop_time_headsign_change;}
+    std::unordered_map<std::string, std::unordered_set<const nt::MetaVehicleJourney*>>
+    get_headsign_mvj() {return headsign_mvj;}
+};
+
+BOOST_AUTO_TEST_CASE(headsign_handler_internal_test) {
```

Comment:
it's not relevant anymore (since you change the heasign_handler outside of the test), but a boost_test inherit from the fixture, so you can use the protected members (so your fixture could have inherited from the handler)